gtdynamics/factors/PoseFactor.h

```diff
@@ -37,10 +38,10 @@ using boost::assign::cref_list_of;
  * Given the joint model, this factor optimizes for the underlying joint axis
  * and the corresponding poses of the parent and child links.
  */
-class PoseFactor : public gtsam::NoiseModelFactor {
+class PoseFactor : public gtsam::ExpressionFactor<gtsam::Vector6> {
```

Comment:
replace with inline function `PoseFactor`
Add a note that this is provided for BW compatibility only, and will in time be replaced with EqualityConstraint

Reply:
Done